src/modules/SpaceCalculation.rb

```diff
@@ -1093,7 +1036,53 @@ def btrfs_used_size(directory)
     # where unit can be one of: "" (none) or "B", "KiB", "MiB", "GiB", "TiB", "PiB"
     # @return [Integer] size in bytes
     def size_from_string(size_str)
-      WFM.call("wrapper_storage", ["ClassicStringToByte", [size_str]])
+      classic_locale = true
+      # Assume bytes by default
+      size_str += "B" unless size_str =~ /[[:alpha:]]/
+      ::Storage.humanstring_to_byte(size_str, classic_locale)
+    end
+
+  private
+
+    # Filesystems to consider while checking the system available space
+    #
+    # @return [Array<Storage::Filesystem>]
+    def target_filesystems
+      filesystems = ::Storage::Filesystem.all(staging_devicegraph).to_a
+      filesystems.select! { |fs| fs.mountpoints.any? { |mp| mp.start_with?("/") } }
+      filesystems.reject! { |fs| TARGET_FS_TYPES_TO_IGNORE.include?(fs.type) }
+      filesystems
+    end
+
+    def staging_devicegraph
+      storage_object.staging
+    end
+
+    def filesystem_size(filesystem)
+      blk_device = filesystem.blk_devices[0]
+      # Only for local fs, NFS not supported yet in libstorage-ng
+      return 0 unless blk_device
+      blk_device.size
+    end
+
+    def filesystem_block_size(filesystem)
+      filesystem.blk_devices[0].region.block_size
+    end
+
+    def filesystem_name(filesystem)
+      filesystem.blk_devices[0].name
+    end
+
+    def filesystem_mountpoint(filesystem)
+      filesystem.mountpoints[0] || ""
+    end
+
+    def type?(type_label, filesystem)
```

Comment:
to be honest as @ancorgs pointed above I really do not see this as part of good design. I know that it is sometimes needed for old code, but for new object oriented code this approach looks horrible. I expect methods like
```
filesystem.type.ext?
filesystem.type.btrfs?
filesystem.type.to_sym # so it can be use in case statement above
```